### PR TITLE
Properly support Postgres table schema

### PR DESCRIPTION
**Pull request by Samjayyy**
Since we are moving director backend to apiserver (go). It doesn't make sense to keep the sql schema there anymore. As a plus we would like to have the same builtin features like pagination for the director tables.
For now we still want the existing director backend to work, so for ease of migration we are keeping the tables in the director schema. The go generation currently does not allow multiple schemas, therefor we made a little change that it also accepts multiple schemas from the same DB.

### CODE REVIEW

**Review by dhoske — COMMENTED** [review]
@xingjianan @tigersoldier Is this util forked from the baseline smallnest-gen? How do we usually make changes here?

File: dbmeta/codegen.go [hunk 1]
```diff
@@ -501,6 +501,7 @@ func (c *Config) CreateContextForTableFile(tableInfo *ModelInfo) map[string]inte
 	var modelInfo = map[string]interface{}{
 		"StructName":         tableInfo.StructName,
 		"TableName":          tableInfo.TableSchemaAndName.TableName,
+		"TableSchema":        tableInfo.TableSchemaAndName.TableSchema,
```
Comment (tigersoldier) [inline dbmeta/codegen.go]: hmm do you need to add this? Why not jus use TableSchemaAndName.Schema?

File: main.go [hunk 2]
```diff
@@ -29,7 +29,8 @@ import (
 var (
 	sqlType          = goopt.String([]string{"--sqltype"}, "mysql", "sql database type such as [ mysql, mssql, postgres, sqlite, etc. ]")
 	sqlConnStr       = goopt.String([]string{"-c", "--connstr"}, "nil", "database connection string")
-	sqlDatabase      = goopt.String([]string{"-d", "--database"}, "nil", "Database to for connection. For Postgres, it's the table's schema name.")
+	sqlDatabase      = goopt.String([]string{"-d", "--database"}, "nil", "Database to for connection. For Postgres, it's the table's schema name. Overrides schemas if also provided")
+	sqlSchemas       = goopt.String([]string{"-s", "--schemas"}, "nil", "If postgres, allows for multiple postgres schemas")
```
Comment (tigersoldier) [inline main.go]: Do we need to add this flag? sqlDatabase is treated as postgres table schema

**Comment by tigersoldier** [comment]
> @xingjianan @tigersoldier Is this util forked from the baseline smallnest-gen? How do we usually make changes here?

@dhoske we made our customization here and didnt' bother merging it to upstream.

File: dbmeta/codegen.go (hunk 1 above)
Comment (Samjayyy) [reply dbmeta/codegen.go]: Just did in analogy with the TableName parameter above. But your suggestion would also work, any preference?

File: main.go (hunk 2 above)
Comment (Samjayyy) [reply main.go]: We could for sure just reuse the same flag too. The idea was to somehow make it a bit more clear when using it.
Any preference?

File: dbmeta/codegen.go (hunk 1 above)
Comment (tigersoldier) [reply dbmeta/codegen.go]: `TableName` was in the original repo. `TableSchema` was added by me. I didn't remove `TableName` for backward compatibility.

File: main.go (hunk 2 above)
Comment (tigersoldier) [reply main.go]: We need to be clear that for postgres`sqlSchema` is replacing `sqlDatabase` if `sqlSchema` is specified. The rest of the code change LGTM.

Can you also update the README?

**Review by tigersoldier — APPROVED** [review]

**Pull request merged**